Lib/mimetypes.py

```diff
@@ -113,6 +113,10 @@ def guess_type(self, url, strict=True):
         Optional `strict' argument when False adds a bunch of commonly found,
         but non-standard types.
         """
+        url = url_or_path
+        if isinstance(url_or_path, os.PathLike):
```

Comment:
You probably don't need the conditional here, `url = os.fspath(url)` also works on strings.